Source/HYPScatterPlot.m

```diff
@@ -28,136 +35,212 @@ - (instancetype)initWithFrame:(CGRect)frame
-        [self drawTextInContext:context rect:rect label:maximumXLabel font:font alignment:NSTextAlignmentRight point:point];
-    }
+    [self updateYAxes];
+    [self updateLabels];
+    [self drawPointsInRect:drawInRect];
 }
 
-- (void)drawTextInContext:(CGContextRef)context rect:(CGRect)rect label:(HYPScatterLabel *)label font:(UIFont *)font alignment:(NSTextAlignment)alignment point:(CGPoint)point
+- (NSArray *)gradientColors
 {
-    //  Note: we can't get a perfect bounding box for the text as the methods that are suppose to do it are buggy:
-    //  http://stackoverflow.com/a/7014352/550393
-    CTFontRef fontRef = CTFontCreateWithName((CFStringRef)font.fontName, font.pointSize, NULL);
-    NSDictionary *attributesDictionary = [NSDictionary dictionaryWithObjectsAndKeys:
-                                          CFBridgingRelease(fontRef), (NSString*)kCTFontAttributeName,
-                                          (id)label.textColor.CGColor, (NSString*)kCTForegroundColorAttributeName,
-                                          nil];
-
-    NSMutableAttributedString *attributedString = [[NSMutableAttributedString alloc] initWithString:label.text attributes:attributesDictionary];
-
-    NSMutableParagraphStyle *paragraphStyle = [[NSMutableParagraphStyle alloc] init];
-    paragraphStyle.alignment = alignment;
-
-    [attributedString addAttribute:NSParagraphStyleAttributeName value:paragraphStyle range:NSMakeRange(0, [label.text length])];
-
-    CGMutablePathRef path = CGPathCreateMutable();
-
-    CTFramesetterRef framesetter = CTFramesetterCreateWithAttributedString((CFAttributedStringRef)attributedString);
-
-    CGRect textRect = CGRectMake(point.x, point.y, CGRectGetMinX(rect), ceilf(font.lineHeight));
-    CGPathAddRect(path, NULL, textRect);
-    CTFrameRef frame = CTFramesetterCreateFrame(framesetter, CFRangeMake(0, [attributedString length]), path, NULL);
-
-    CFRelease(framesetter);
-    CFRelease(path);
+    return [NSArray arrayWithObjects:
+            (id)self.yAxisEndGradient.CGColor,
+            (id)self.yAxisMidGradient.CGColor,
+            (id)self.yAxisEndGradient.CGColor, nil];
+}
 
-    CTFrameDraw(frame, context);
+- (void)updateLabels
+{
+    self.minimumXLabel.attributedText = [self.dataSource minimumXValueFormattedInScatterPlotView:self];
+    self.maximumXLabel.attributedText = [self.dataSource maximumXValueFormattedInScatterPlotView:self];
+    self.minimumYLabel.attributedText = [self.dataSource minimumYValueFormattedInScatterPlotView:self];
+    self.maximumYLabel.attributedText = [self.dataSource maximumYValueFormattedInScatterPlotView:self];
 
-    CFRelease(frame);
+    if ([self.dataSource respondsToSelector:@selector(averageYValueFormattedInScatterPlotView:)]) {
+        self.averageYLabel.attributedText = [self.dataSource averageYValueFormattedInScatterPlotView:self];
+    }
 }
 
-- (void)drawAxes:(CGContextRef)context rect:(CGRect)rect
+- (void)updateYAxes
```

Comment:
Is "axes" intentional? Maybe "axis" instead? (I was surprised to learn that "axes" is the plural form for "axis" though)